modules/control/controller/lat_controller.cc

```diff
@@ -269,8 +270,66 @@ Status LatController::ComputeControlCommand(
     ControlCommand *cmd) {
   VehicleStateProvider::instance()->set_linear_velocity(chassis->speed_mps());
 
+  auto target_tracking_trajectory = *planning_published_trajectory;
+
+  if (FLAGS_use_navigation_mode) {
+    auto time_stamp_diff =
+        planning_published_trajectory->header().timestamp_sec()
+        - current_trajectory_timestamp_;
+
+    auto curr_vehicle_x = localization->pose().position().x();
+    auto curr_vehicle_y = localization->pose().position().y();
+
+    double curr_vehicle_heading = 0.0;
+    const auto &orientation = localization->pose().orientation();
+    if (localization->pose().has_heading()) {
+      curr_vehicle_heading = localization->pose().heading();
+    } else {
+      curr_vehicle_heading =
+          common::math::QuaternionToHeading(orientation.qw(), orientation.qx(),
+                                    orientation.qy(), orientation.qz());
+    }
+
+    // new plannng trajectory
+    if (time_stamp_diff > 1.0e-6) {
+      init_vehicle_x_ = curr_vehicle_x;
+      init_vehicle_y_ = curr_vehicle_y;
+      init_vehicle_heading_ = curr_vehicle_heading;
+
+      current_trajectory_timestamp_ =
+          planning_published_trajectory->header().timestamp_sec();
+    } else {
+      auto x_diff = curr_vehicle_x - init_vehicle_x_;
```

Comment:
@YajiaZhang We should add extra evaluation of std_dev of GPS/Localization here to determine whether we do the frame rotation here right? -- Only do this rotation when we have accurate enough localization signal.

Reply:
This is equivalent to the case that control in absolute map frame with noisy localization. If we don't do the check in that case, we do not need to do this either. Also, the std dev can jump over time as well which will lead to unnecessary switching between time matching and position matching.